roboteam_utils/src/utils/Field.cpp

```diff
@@ -13,7 +13,17 @@ bool Field::operator==(const Field &other) const {
         && this->leftDefenseArea == other.leftDefenseArea
         && this->rightDefenseArea == other.rightDefenseArea
         && this->leftGoalArea == other.leftGoalArea
-        && this->rightGoalArea == other.rightGoalArea;
+        && this->rightGoalArea == other.rightGoalArea
+        && this->bottomLeftGrid == other.bottomLeftGrid
+        && this->bottomMidGrid == other.bottomMidGrid
+        && this->bottomRightGrid == other.bottomRightGrid
+        && this->middleLeftGrid == other.middleLeftGrid
+        && this->middleMidGrid == other.middleMidGrid
+        && this->middleRightGrid == other.middleRightGrid
+        && this->topLeftGrid == other.topLeftGrid
+        && this->topMidGrid == other.topMidGrid
+        && this->topRightGrid == other.topRightGrid;
```

Comment:
Yes, agree on this too, but for tests it is convenient to check if the grids also match when comparing fields. On top of that, it might be misleading to use the operator== function for comparing field objects without checking the grids. But this will all be solved when the grids are no longer part of the Field class. TODO!